kie-drools-wb/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/docks/AuthoringWorkbenchDocks.java

```diff
@@ -65,14 +68,28 @@ public void setup(String authoringPerspectiveIdentifier, PlaceRequest projectExp
     }
 
     public void onContextChange(@Observes DataModelerWorkbenchContextChangeEvent contextEvent) {
+        handleDocks();
+    }
+
+    private void handleDocks() {
         DataModelerContext context = dataModelerWBContext.getActiveContext();
-        if (shouldDisplayWestDocks(context)) {
+        if (!dataModelerIsHidden && shouldDisplayWestDocks(context)) {
             uberfireDocks.enable(UberfireDockPosition.EAST, authoringPerspectiveIdentifier);
         } else {
             uberfireDocks.disable(UberfireDockPosition.EAST, authoringPerspectiveIdentifier);
         }
     }
 
+    public void onDataModelerWorkbenchFocusEvent( @Observes DataModelerWorkbenchFocusEvent event ) {
+        if ( !event.isFocused() ) {
```

Comment:
I love your double negatives :)

"if it doesn't have the focus then it is hidden" vs "if it does have the focus it is visible"..

NP.. will merge anyway.